azkaban-common/src/main/java/azkaban/utils/YarnUtils.java

```diff
@@ -30,6 +38,38 @@ public class YarnUtils {
   private static final String YARN_CONF_DIRECTORY_PROPERTY = "env.YARN_CONF_DIR";
   private static final String YARN_CONF_FILENAME = "yarn-site.xml";
 
+  public static final EnumSet<YarnApplicationState> YARN_APPLICATION_ALIVE_STATES = EnumSet.of(
+      YarnApplicationState.NEW,
+      YarnApplicationState.NEW_SAVING,
+      YarnApplicationState.SUBMITTED,
+      YarnApplicationState.ACCEPTED,
+      YarnApplicationState.RUNNING
+  );
+
+  /**
+   * Use the yarnClient to query the unfinished yarn applications, then use the yarnClient to kill
+   * them sequentially
+   *
+   * @param yarnClient the yarnClient already connects to the cluster
+   * @param flowExecID the azkaban flow execution id whose yarn applications needs to be killed
+   * @return the set of all to-be-killed (alive) yarn applications' IDs
+   * @throws IOException   for RPC issue
+   * @throws YarnException for YARN server issue
+   */
+  public static Set<String> getAllAliveAppIDsByExecID(final YarnClient yarnClient,
```

Comment:
You would like to return the ApplicationReport from this method and create another method for application id. Since we need to get the user from ApplicationReport as well, if we write it in the current way, we need to have another call to get the user.